### PR TITLE
Tournaments: track scores of in-progress matches

### PR DESCRIPTION
**Pull request by sirDonovan**
Send with bracket data in tournament updates to be displayed in the client-side bracket.

### CODE REVIEW

File: battle-engine.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ process.on('message', function (message) {
 	if (data[1] === 'init') {
 		if (!Battles[data[0]]) {
 			try {
-				Battles[data[0]] = Battle.construct(data[0], data[2], data[3]);
+				Battles[data[0]] = Battle.construct(data[0], data[2], data[3], data[4]);
```
Comment (Zarel) [inline battle-engine.js]: ಠ_ಠ

You do _not_ need this.

Comment (sirDonovan) [inline battle-engine.js]: Ok, removed that part.

**Comment by Slayer95** [comment]
Won't this leak information in pre-BW battles (no Team Preview)?

**Comment by DanUgore** [comment]
What do you mean? Scores have always been reported as pokemon remaining.

**Comment by DanUgore** [comment]
Oh Pre-BW my mistake.

**Comment by sirDonovan** [comment]
I think it will be confusing for users if we change the format for scores, maybe we can just omit this feature for pre-BW tournaments?

**Comment by Zarel** [comment]
Why do we need scores, anyway?

**Comment by Zarel** [comment]
They're not a particularly good measure of who's winning.

**Comment by DanUgore** [comment]
Some see it that way http://puu.sh/gbRzk.txt

**Comment by sirDonovan** [comment]
They wouldn't be used to see who's winning or losing, but to give players a better idea of how much time they have until their next match. They would also allow other users to estimate the remaining time of a certain match or round of the tournament by looking over the bracket instead of joining battles.

**Comment by Zarel** [comment]
>Some see it that way http://puu.sh/gbRzk.txt

And we both know that's dumb. :|

>They wouldn't be used to see who's winning or losing, but to give players a better idea of how much time they have until their next match. They would also allow other users to estimate the remaining time of a match or round of a tournament by looking over the bracket instead of joining battles.

There's probably better ways of doing this that doesn't involve this kind of overhead.

File: simulator.js [hunk 2]
```diff
@@ -194,6 +194,7 @@ var Battle = (function () {
 			break;
 
 		case 'score':
+			if (!this.room.tour) break;
```
Comment (kotarou3) [inline simulator.js]: This shouldn't be nessasary.

**Comment by sirDonovan** [comment]
@Zarel How about this method?

**Comment by Zarel** [comment]
I still disagree with the idea of tracking scores at all.

**Comment by sirDonovan** [comment]
May I ask why?

**Comment by Zarel** [comment]
I already told you. It doesn't accurately measure who's likely to win or how long the battle will go on for. It's only an inaccurate proxy for either.

**Comment by TheImmortal** [comment]
I agree with Zarel.

**Comment by Zarel** [comment]
I'm going to close this for now. I'll reopen if you can convince me that there's no better way to convey the information and the information is important to convey.

**Pull request closed without merging**